docs/kb/integrations/customize-semgrep-precommit.md

```diff
@@ -0,0 +1,71 @@
+---
+description: Understand how to customize Semgrep's behavior when using it with pre-commit.
+tags:
+  - Extensions
+---
+
+# Customize Semgrep in pre-commit
+
+Semgrep offers two standard pre-commit hooks [outlined here](/docs/extensions/overview/#pre-commit).
+
+* The `semgrep` hook is optimized to run with a provided set of rules, either local rules or rules from the Semgrep Registry.
+* The `semgrep-ci` hook is optimized to run with the rules from your organization in the Semgrep AppSec Platform, without sending results to the platform, since results from pre-commit are temporary and not linked to the final code in the repository.
+
+You can also create your own Semgrep hooks for pre-commit if you have particular needs or preferences, such as if you want users to see the output of the Semgrep scan even if it passed, or if you want to limit the scan to a particular Semgrep product.
+
+## Show the scan output
+
+By default, pre-commit does not show the scan output if the hook passes. If you're using the `semgrep-ci` hook, but want your developers to see the output even if non-blocking findings are found, you can set the `verbose` option to print the output. To limit output to findings only (no scan information or diagnostics), redirect `stderr` to `/dev/null`, or use `--quiet`.
```

Comment:
what if I'm just using the semgrep hook, not semgrep-ci? Do I see the output then?

Reply:
This is just written as an example, but I'll make that clearer in the language.